// @coreui/angular/esm2015/lib/sidebar/app-sidebar-nav/app-sidebar-nav-dropdown.component.js
import { __decorate, __metadata } from "tslib";
import { Component, Input } from '@angular/core';
import { SidebarNavHelper } from '../app-sidebar-nav.service';
let AppSidebarNavDropdownComponent = class AppSidebarNavDropdownComponent {
    constructor(helper) {
        this.helper = helper;
    }
};
AppSidebarNavDropdownComponent.ctorParameters = () => [
    { type: SidebarNavHelper }
];
__decorate([
    Input(),
    __metadata("design:type", Object)
], AppSidebarNavDropdownComponent.prototype, "item", void 0);
AppSidebarNavDropdownComponent = __decorate([
    Component({
        selector: 'app-sidebar-nav-dropdown, cui-sidebar-nav-dropdown',
        template: `
    <a class="nav-link nav-dropdown-toggle"
       appNavDropdownToggle
       [appHtmlAttr]="item.attributes">
      <i *ngIf="helper.hasIcon(item)" [ngClass]="item | appSidebarNavIcon"></i>
      <ng-container>{{item.name}}</ng-container>
      <span *ngIf="helper.hasBadge(item)" [ngClass]="item | appSidebarNavBadge">{{ item.badge.text }}</span>
    </a>
    <app-sidebar-nav-items
      class="nav-dropdown-items"
      [items]="item.children">
    </app-sidebar-nav-items>
  `,
        providers: [SidebarNavHelper],
        styles: ['.nav-dropdown-toggle { cursor: pointer; }',
            '.nav-dropdown-items { display: block; }']
    }),
    __metadata("design:paramtypes", [SidebarNavHelper])
], AppSidebarNavDropdownComponent);
export { AppSidebarNavDropdownComponent };
